plugins/anomaly-detection/AnomalyDetection.py

```diff
@@ -0,0 +1,615 @@
+            "throughputStandardDeviation"))
+    plotDF2 = plotDF.groupby("sourceIP", "sourceTransportPort",
+                             "destinationIP", "destinationTransportPort",
+                             "protocolIdentifier", "flowStartSeconds").agg(
+        f.collect_list(f.struct(["Diff_Secs", "max(throughput)"])))
+    plotDF2 = plotDF2.withColumnRenamed(
+        "collect_list(struct(Diff_Secs, max(throughput)))",
+        "Diff_Secs, Throughput")
+    plotDF3 = plotDF.groupby("sourceIP", "sourceTransportPort",
+                             "destinationIP", "destinationTransportPort",
+                             "protocolIdentifier", "flowStartSeconds").agg(
+        f.collect_list("flowEndSeconds"))
+    plotDF3 = plotDF3.withColumnRenamed(
+        "collect_list(flowEndSeconds)",
+        "flowEndSeconds")
+
+    # Joining Two different Groupby DFs
+    plotDF4 = plotDF3.join(plotDF1,
+                           ["sourceIP", "sourceTransportPort", "destinationIP",
+                            "destinationTransportPort", "protocolIdentifier",
+                            "flowStartSeconds"])
+    plotDF5 = plotDF4.join(plotDF2,
+                           ["sourceIP", "sourceTransportPort", "destinationIP",
+                            "destinationTransportPort", "protocolIdentifier",
+                            "flowStartSeconds"])
+    if algo_type == "EWMA":
+        ret_plot = plot_anomaly(spark, plotDF5, algo_type, calculate_ewma,
+                                calculate_ewma_anomaly, tad_id_input)
+    elif algo_type == "ARIMA":
+        ret_plot = plot_anomaly(spark, plotDF5, algo_type, calculate_arima,
+                                calculate_arima_anomaly, tad_id_input)
+    elif algo_type == "DBSCAN":
+        ret_plot = plot_anomaly(spark, plotDF5, algo_type, calculate_dbscan,
+                                calculate_dbscan_anomaly, tad_id_input)
+    return spark, ret_plot
+
+
+def write_anomaly_detection_result(
+        result_df, db_jdbc_address, result_table_name, tad_id_input):
+    if not tad_id_input:
+        tad_id = str(uuid.uuid4())
+    else:
+        tad_id = tad_id_input
+
+    result_df.write.mode("append").format("jdbc").option(
+        "driver", "ru.yandex.clickhouse.ClickHouseDriver").option(
+        "url", db_jdbc_address).option(
+        "user", os.getenv("CH_USERNAME")).option(
+        "password", os.getenv("CH_PASSWORD")).option(
+        "dbtable", result_table_name).save()
+    return tad_id
+
+
+def main():
+    db_jdbc_address = (
+        "jdbc:clickhouse://clickhouse-clickhouse.flow-visibility.svc:8123")
+    result_table_name = "default.tadetector"
+    algo_type = ""
+    start_time = ""
+    end_time = ""
+    tad_id_input = None
+    help_message = """
+    Start the policy recommendation spark job.
+        Options:
+        -h, --help: Show help message.
+        -t, --type=EWMA: Type argument describes the anomaly detection Algo 
```

Comment:
Let's rename this argument from `type` to `algorithm`

Reply:
algorithm would be too much to write for user, do you think we can make it algo?

Reply:
sure, sounds good

Reply:
ack